src/runtime/pyobject.cs

```diff
@@ -1133,10 +1133,26 @@ public override bool TryBinaryOperation(BinaryOperationBinder binder, Object arg
                     result = null;
                     return false;
             }
-            result = new PyObject(res);
+            result = CheckNone(new PyObject(res));
+
             return true;
         }
 
+        // Workaround for https://bugzilla.xamarin.com/show_bug.cgi?id=41509
+        // See https://github.com/pythonnet/pythonnet/pull/219
+        private static object CheckNone(PyObject pyObj)
+        {
+            if (pyObj != null)
+            {
+                if (pyObj.obj == Runtime.PyNone)
```

Comment:
you could and even should imho, but then you are triggering the mono bug :). The problem is later. when you save such a none object in a dynamic and want to resolve it to a string (for example). This will not work as long as the mono bug exists. In fact one of my tests discovered this glitch